### PR TITLE
OpenGL: Fix DisplayTransfer accel when input width != output width

### LINKED ISSUE #2246: Pokémon Omega Ruby/Alpha Saphire ‒ Top screen garbling bug
This happens when you found a wild pokemon and it shows the transition effect before starting the battle
Using *Hardware Renderer* displays this during transition:
Pixelated graphic

Nothing useful in log:
```
Service.APT <Warning> core\hle\service\apt\apt.cpp:SetScreenCapPostPermission:416: (STUBBED) screen_capture_post_permission=3
Service.CFG <Warning> core\hle\service\cfg\cfg.cpp:GenHashConsoleUnique:177: (STUBBED) called app_id_salt=0x55D
Service.FRD <Warning> core\hle\service\frd\frd.cpp:GetMyFriendKey:91: (STUBBED) called
```
Switching to *Sofware Renderer* will playback the animation exactly like the actual hardware behaviour.

Tested this on commit 3174bfd

**OS:** _Windows 10 build 14393.447 (codename Redstone 1)_
**GPU:** _nVidia GT 610 2GB DDR3 (Asus)_
**Driver:** _Version 375.95_
**OpenGL:** _Version 4.5_

### PR DESCRIPTION
**Pull request by yuriks**
First commit is the actual fix. Others are small cleanups/ticking-time-bomb bugfixes.
Fixes #2246, #2261.

### CODE REVIEW

**Comment by yuriks** [comment]
cc @tfarley in case he wants to give some input on how I missed a crucial details in the implementation here and this is actually the wrong way to fix it. :)

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 1]
```diff
@@ -291,6 +289,10 @@ CachedSurface* RasterizerCacheOpenGL::GetSurface(const CachedSurface& params, bo
 
     MICROPROFILE_SCOPE(OpenGL_SurfaceUpload);
 
+    if (params.is_tiled) {
+        ASSERT(params.stride == 0);
+    }
```
Comment (Subv) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: A comment about this would be nice.
Why?

Comment (yuriks) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: This *is* the comment. ;) (I spent some time chasing this lead before realizing that stride only makes sense for linear images.)

**Comment by Subv** [comment]
LGTM

**Pull request merged**